storage-proofs/src/zigzag_graph.rs

```diff
@@ -11,18 +11,78 @@ use crate::settings;
 
 pub const DEFAULT_EXPANSION_DEGREE: usize = 8;
 
-// Cache of node's parents.
-pub type ParentCache = HashMap<usize, Vec<usize>>;
+lazy_static! {
+    // This parents cache is currently used for the *expanded parents only*, generated
+    // by the expensive Feistel operations in the ZigZag, it doesn't contain the
+    // "base" (in the `Graph` terminology) parents, which are cheaper to compute.
+    static ref PARENT_CACHE: Arc<RwLock<HashMap<String, ParentCache>>> = Arc::new(RwLock::new(HashMap::new()));
```

Comment:
It may be beneficial to document how the hashmap is keyed, it's not obvious to me that the `String` key is the `ZigZagGraph`'s id.